src/Uno.UI/UI/Xaml/DependencyObjectCollection.cs

```diff
@@ -25,7 +26,29 @@ public partial class DependencyObjectCollectionBase : DependencyObject
 	public partial class DependencyObjectCollection<T> : DependencyObjectCollectionBase, IList<T>, IEnumerable<T>, IEnumerable, IObservableVector<T>
 		where T : DependencyObject
 	{
-		public event VectorChangedEventHandler<T> VectorChanged;
+		private object _vectorChangedHandlersLock = new();
+
+		// Explicit handlers list to avoid the cost of multicast delegates handling
+		private List<VectorChangedEventHandler<T>> _vectorChangedHandlers;
+
+		public event VectorChangedEventHandler<T> VectorChanged
+		{
+			add
+			{
+				lock (_vectorChangedHandlersLock)
+				{
+					(_vectorChangedHandlers ??= new()).Add(value);
+				}
+			}
+
+			remove
+			{
+				lock (_vectorChangedHandlersLock)
+				{
+					(_vectorChangedHandlers ??= new()).Remove(value);
```

Comment:
This is a change in behavior I believe. `List<T>.Remove` will remove the first occurrence, while regular events removes the last occurrence.

So, if you do:

```
collection.VectorChanged += firstHandler;
collection.VectorChanged += secondHandler;
collection.VectorChanged += firstHandler; // EXACTLY the same instance.
collection.VectorChanged -= firstHandler;

// something that triggers the event.

// which handler is invoked first?
```

Reply:
For this one, it might be good to also verify what is the behavior of WinUI, as I'm not very sure what WinUI behavior here is. I haven't verified.

Reply:
It's removing the last. I've adjusted it.